src/Common/Commands.Common/MetricHelper.cs

```diff
@@ -52,6 +54,24 @@ public class MetricHelper
         /// </summary>
         private readonly object _lock = new object();
 
+        private string _hashMacAddress = string.Empty;
+
+        private string HashMacAddress
+        {
+            get
+            {
+                if (_hashMacAddress == string.Empty)
+                {
+                    var macAddress = NetworkInterface.GetAllNetworkInterfaces()
+                        .FirstOrDefault(nic => nic.OperationalStatus == OperationalStatus.Up)?
+                        .GetPhysicalAddress().ToString();
```

Comment:
You actually need an elvis operator preceding the `ToString()` since the preceding code can evaluate to `null`.

```csharp
var macAddress = NetworkInterface.GetAllNetworkInterfaces()
    .FirstOrDefault(nic => nic.OperationalStatus == OperationalStatus.Up)?
    .GetPhysicalAddress()?.ToString();
```